torch/csrc/autograd/engine.cpp

```diff
@@ -816,8 +823,29 @@ variable_list Engine::execute_with_graph_task(
     // The owning thread start to drive the engine execution with the GraphTask
     // that has already been pushed to the current CPU thread's ready_queue
     lock.unlock();
-    thread_main(nullptr, false);
-
+    if (async_mode) {
+      std::function<void()> handle_work = [this, &graph_task, &handle_work]() {
+        while(!local_ready_queue->empty()) {
+          auto task = local_ready_queue->pop();
+          execute_node_task(task);
+        }
+        // Check if we've completed execution.
+        if (graph_task_completed(graph_task)) {
+          // We don't need to explicitly notify the owner thread, since
+          // 'mark_graph_task_completed' would mark the Future as completed and this
+          // would notify the owner thread that the task has been completed.
+          mark_graph_task_completed(graph_task);
+        } else {
+          // schedule a continuation
+          std::cout << "start scheduling a continuation" << std::endl;
```

Comment:
nit: Use LOG(DEBUG) or VLOG(1) instead? There might be quite a bit of log spew due to this.